diff --git a/ROS2_HUMBLE/src/stararm102_description/CMakeLists.txt b/ROS2_HUMBLE/src/stararm102_description/CMakeLists.txt
index bcdffec..f5411d4 100644
--- a/ROS2_HUMBLE/src/stararm102_description/CMakeLists.txt
+++ b/ROS2_HUMBLE/src/stararm102_description/CMakeLists.txt
@@ -3,8 +3,6 @@ project(stararm102_description)
 
 find_package(ament_cmake REQUIRED)
 
-ament_package()
-
 if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/launch")
   install(
     DIRECTORY launch
@@ -12,5 +10,6 @@ if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/launch")
     PATTERN "setup_assistant.launch" EXCLUDE)
 endif()
 
-install(DIRECTORY config DESTINATION share/${PROJECT_NAME})
+install(DIRECTORY meshes urdf DESTINATION share/${PROJECT_NAME})
-install(FILES .setup_assistant DESTINATION share/${PROJECT_NAME})
+
+ament_package()
diff --git a/ROS2_HUMBLE/src/stararm102_description/package.xml b/ROS2_HUMBLE/src/stararm102_description/package.xml
index 8a3fcfc..2648b66 100644
--- a/ROS2_HUMBLE/src/stararm102_description/package.xml
+++ b/ROS2_HUMBLE/src/stararm102_description/package.xml
@@ -7,18 +7,20 @@
   <maintainer email="12461360+nyancos@user.noreply.gitee.com">nyancos</maintainer>
   <license>MIT</license>
 
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
   <test_depend>ament_copyright</test_depend>
   <test_depend>ament_flake8</test_depend>
   <test_depend>ament_pep257</test_depend>
   <test_depend>python3-pytest</test_depend>
 
-    <depend>urdf</depend>
-    <build_depend>launch_ros</build_depend>
-    <exec_depend>launch_ros</exec_depend>
-    <exec_depend>robot_state_publisher</exec_depend>
-    <exec_depend>joint_state_publisher_gui</exec_depend>
+  <depend>urdf</depend>
+  <build_depend>launch_ros</build_depend>
+  <exec_depend>launch_ros</exec_depend>
+  <exec_depend>robot_state_publisher</exec_depend>
+  <exec_depend>joint_state_publisher_gui</exec_depend>
 
   <export>
-    <build_type>ament_python</build_type>
+    <build_type>ament_cmake</build_type>
   </export>
 </package>
diff --git a/ROS2_HUMBLE/src/stararm102_description/urdf/stararm102_description.urdf b/ROS2_HUMBLE/src/stararm102_description/urdf/stararm102_description.urdf
index 7af5b15..23ff77f 100644
--- a/ROS2_HUMBLE/src/stararm102_description/urdf/stararm102_description.urdf
+++ b/ROS2_HUMBLE/src/stararm102_description/urdf/stararm102_description.urdf
@@ -106,8 +106,8 @@
     <axis
       xyz="0 0 -1" />
     <limit
-      lower="-2.27"
-      upper="2.27"
+      lower="-1.919862177"
+      upper="1.919862177"
       effort="100"
-      velocity="1" />
+      velocity="3.560471674" />
   </joint>
@@ -164,8 +164,8 @@
     <axis
       xyz="0 0 -1" />
     <limit
-      lower="0"
-      upper="3.14"
+      lower="0"
+      upper="3.141592654"
       effort="100"
-      velocity="1" />
+      velocity="3.560471674" />
   </joint>
@@ -221,9 +221,9 @@
       link="link3" />
     <axis
       xyz="0 0 -1" />
    <limit
-      lower="-3.14"
-      upper="0"
+      lower="-4.712388980"
+      upper="0"
       effort="100"
-      velocity="1" />
+      velocity="3.560471674" />
   </joint>
@@ -280,8 +280,8 @@
     <axis
       xyz="0 0 -1" />
     <limit
-      lower="-1.57"
-      upper="2.2"
+      lower="-1.570796327"
+      upper="1.570796327"
       effort="100"
-      velocity="1" />
+      velocity="3.560471674" />
   </joint>
@@ -338,8 +338,8 @@
     <axis
       xyz="0 0 -1" />
     <limit
-      lower="-1.2"
-      upper="1.2"
+      lower="-1.134464014"
+      upper="1.134464014"
       effort="100"
-      velocity="1" />
+      velocity="3.560471674" />
   </joint>
@@ -396,8 +396,8 @@
     <axis
       xyz="0 0 -1" />
     <limit
-      lower="-3.14"
-      upper="3.14"
+      lower="-2.617993878"
+      upper="2.617993878"
       effort="100"
-      velocity="1" />
+      velocity="3.560471674" />
   </joint>
@@ -453,9 +453,9 @@
       link="link7_left" />
     <axis
       xyz="0 0 -1" />
-    <limit
-      lower="0"
-      upper="1.57"
+    <limit
+      lower="0"
+      upper="1.570796327"
       effort="5"
-      velocity="3.14" />
+      velocity="3.560471674" />
   </joint>
@@ -511,11 +511,17 @@
       link="link7_right" />
     <axis
       xyz="0 0 -1" />
-    <limit
-      lower="-1.57"
+    <limit
+      lower="-1.570796327"
       upper="0"
       effort="5"
-      velocity="3.14" />
+      velocity="3.560471674" />
-    <mimic joint="joint7_left" multiplier="-1" />
-  </joint>
-</robot>
+    <mimic joint="joint7_left" multiplier="-1" />
+  </joint>
+  <link name="tcp_link" />
+  <joint name="tcp_joint" type="fixed">
+    <origin xyz="0 0 0" rpy="0 0 0" />
+    <parent link="link6" />
+    <child link="tcp_link" />
+  </joint>
+</robot>
diff --git a/ROS2_HUMBLE/src/stararm102_moveit_config/config/kinematics.yaml b/ROS2_HUMBLE/src/stararm102_moveit_config/config/kinematics.yaml
index 1021bc2..18192bf 100644
--- a/ROS2_HUMBLE/src/stararm102_moveit_config/config/kinematics.yaml
+++ b/ROS2_HUMBLE/src/stararm102_moveit_config/config/kinematics.yaml
@@ -1,4 +1,3 @@
 arm:
-  kinematics_solver: kdl_kinematics_plugin/KDLKinematicsPlugin
-  kinematics_solver_search_resolution: 0.0050000000000000001
+  kinematics_solver: trac_ik_kinematics_plugin/TRAC_IKKinematicsPlugin
   kinematics_solver_timeout: 0.0050000000000000001
