diff --git a/ROS2_HUMBLE/src/stararm102_moveit_config/config/stararm102_description.ros2_control.xacro b/ROS2_HUMBLE/src/stararm102_moveit_config/config/stararm102_description.ros2_control.xacro
--- a/ROS2_HUMBLE/src/stararm102_moveit_config/config/stararm102_description.ros2_control.xacro
+++ b/ROS2_HUMBLE/src/stararm102_moveit_config/config/stararm102_description.ros2_control.xacro
@@ -2,73 +2,58 @@
 <robot name="stararm102_description">
   <ros2_control name="StarArm102System" type="system">
     <hardware>
-      <plugin>mock_components/GenericSystem</plugin>
+      <plugin>joint_state_topic_hardware_interface/JointStateTopicSystem</plugin>
+      <param name="joint_commands_topic">/stararm102/joint_commands</param>
+      <param name="joint_states_topic">/stararm102/joint_states</param>
     </hardware>
     <joint name="joint1">
       <command_interface name="position">
         <param name="min">-2.27</param>
         <param name="max">2.27</param>
       </command_interface>
-      <state_interface name="position"/>
-      <state_interface name="velocity"/>
+      <state_interface name="position"><param name="initial_value">0</param></state_interface>
     </joint>
     <joint name="joint2">
       <command_interface name="position">
         <param name="min">-1.57</param>
         <param name="max">1.57</param>
       </command_interface>
-      <state_interface name="position"/>
-      <state_interface name="velocity"/>
+      <state_interface name="position"><param name="initial_value">0</param></state_interface>
     </joint>
     <joint name="joint3">
       <command_interface name="position">
         <param name="min">-1.57</param>
         <param name="max">1.57</param>
       </command_interface>
-      <state_interface name="position"/>
-      <state_interface name="velocity"/>
+      <state_interface name="position"><param name="initial_value">-0.05235987755982989</param></state_interface>
     </joint>
     <joint name="joint4">
       <command_interface name="position">
         <param name="min">-1.57</param>
         <param name="max">1.57</param>
       </command_interface>
-      <state_interface name="position"/>
-      <state_interface name="velocity"/>
+      <state_interface name="position"><param name="initial_value">0</param></state_interface>
     </joint>
    <joint name="joint5">
      <command_interface name="position">
-        <param name="min">-1.57</param>
-        <param name="max">1.57</param>
+        <param name="min">-2.27</param>
+        <param name="max">2.27</param>
      </command_interface>
-      <state_interface name="position"/>
-      <state_interface name="velocity"/>
+      <state_interface name="position"><param name="initial_value">0</param></state_interface>
     </joint>
     <joint name="joint6">
       <command_interface name="position">
         <param name="min">-2.27</param>
         <param name="max">2.27</param>
       </command_interface>
-      <state_interface name="position"/>
-      <state_interface name="velocity"/>
+      <state_interface name="position"><param name="initial_value">0</param></state_interface>
     </joint>
    <joint name="joint7_left">
      <command_interface name="position">
         <param name="min">0</param>
         <param name="max">1.57</param>
      </command_interface>
-      <state_interface name="position"/>
-      <state_interface name="velocity"/>
-    </joint>
-    <joint name="joint7_right">
-      <command_interface name="position">
-        <param name="min">-1.57</param>
-        <param name="max">0</param>
-      </command_interface>
-      <state_interface name="position"/>
-      <state_interface name="velocity"/>
-      <param name="mimic">joint7_left</param>
-      <param name="multiplier">-1</param>
+      <state_interface name="position"><param name="initial_value">0</param></state_interface>
     </joint>
   </ros2_control>
 </robot>
